src/utils/activity-object-helper.ts

```diff
@@ -15,12 +16,20 @@ export class ActivityObjectHelper {
     object: ActivityObject | Link | URL | Array<ActivityObject | URL>
   ): string | undefined {
     const lang = navigator.language.substr(0, 2);
-    if (ActivityObjectHelper.hasProperty(object, "name")) {
+    if (ActivityObjectHelper.hasProperty(object, "name") && object.name) {
       return (object as ActivityObject).name;
     } else if (ActivityObjectHelper.hasProperty(object, "nameMap")) {
-      return (object as RdfLangString).nameMap[lang];
-    } else if (ActivityObjectHelper.hasProperty(object, "map")) {
-      return (object as RdfLangString).map[lang];
+      const rdfLangStringObject = object as RdfLangString;
+
+      if (rdfLangStringObject.nameMap) {
+        return rdfLangStringObject.nameMap[lang];
+      }
+    } else if (ActivityObjectHelper.hasProperty(object, "preferredUsername")) {
+      const actorObject = object as Actor;
+
+      if (actorObject.preferredUsername) {
```

Comment:
Here again, hasProperty has ensured that the preferredUsername exists. This if condition is therefore unnecessary